### PR TITLE
🐛 Expire cache fix for server-side decisions

### LINKED ISSUE #25856: amp-consent expireCache is not working as intended

### PR DESCRIPTION
**Pull request by micajuine-ho**
Fixes #25856

Publishers can now always rely on server side decisions:
```
{
    expireCache: true,
    consentStateValue: accept/reject/unknown
}
```

### CODE REVIEW

File: extensions/amp-consent/0.1/amp-consent.js [hunk 1]
```diff
@@ -548,8 +548,10 @@ export class AmpConsent extends AMP.BaseElement {
       const expireCache = response['expireCache'];
       if (expireCache) {
         this.consentStateManager_.setDirtyBit();
-        // Decision from promptUI takes precedence over consent decision from response
-      } else if (
+      }
+
+      // Decision from promptUI takes precedence over consent decision from response
+      if (
```
Comment (lannka) [inline extensions/amp-consent/0.1/amp-consent.js]: will this override `setDirtyBit` ?

Comment (micajuine-ho) [reply extensions/amp-consent/0.1/amp-consent.js]: No, it will update the values with the dirty bit, which will clear the values.

Comment (lannka) [reply extensions/amp-consent/0.1/amp-consent.js]: I'm a bit confused by this code:
https://github.com/ampproject/amphtml/blob/63fa1d05386d20ff6135a42000cf9d3036b253cd/extensions/amp-consent/0.1/consent-state-manager.js#L296-L312

Just to be safe, shall we just move `setDirtyBit` to after the `updateCacheIfNotNull_` call

Comment (zhouyx) [reply extensions/amp-consent/0.1/amp-consent.js]: `setDirtyBit` sets a local value that will be used when writing value to localStorage. So it's fine to set the dirtyBit earlier. 

However there's a performance concern here. Setting dirtyBit requires a localStorage read that could be combined with the write right afterwards. I'm not sure what the save can be, but worth improving the code later.

Comment (micajuine-ho) [reply extensions/amp-consent/0.1/amp-consent.js]: Good point, definitely worth looking into later.

**Comment by micajuine-ho** [comment]
> I haven't tested manually, but I believe If we want to treat consentInfo from server and client differently (have expireCache set dirtyBit to consentInfo from server) we need to updateConsentInstance with `opt_systemUpdate=true` here. Please verify. Thanks

I can confirm that this is the current state and that adding `opt_systemUpdate=true` will separate client and server `consentInfo` (which is what we want).

**Comment by zhouyx** [comment]
Note from offline discussions:

`expireCache` is always going to remove the stored value on next user visit despite where the decision come from (from user action from prompt or from `checkConsentHref` response).

**Pull request merged**